source/screenExplorer.py

```diff
@@ -1,8 +1,68 @@
+#screenExplorer.py
+#A part of NonVisual Desktop Access (NVDA)
+#Copyright (C) 2012-2018 NV Access Limited, Joseph Lee
+#This file is covered by the GNU General Public License.
+#See the file COPYING for more details.
+
+"""Module which implements various utilities for screen exploration.
+This includes touchscreen movement as well as handling coordinate announcement for mouse and objects.
+"""
+
 import api
 import tones
 import controlTypes
 import textInfos
 import speech
+import screenBitmap
+import config
+import mouseHandler
+import wx
+
+def playLocationCoordinates(x, y, screenWidth, screenHeight, screenMinPos=None, detectBrightness=True,blurFactor=0):
+	"""Plays a tone indicating the XY-coordinate for the given location. This works for keyboard-based object navigation/focus movement, mouse movement and during touch hover gesture.  
+	@param obj: the coordinate for the location and the screen resolution (typically desktopObject.location)
+	@type obj: int
+	"""
+	# Have a fake wx.Point in case min pos is not defined on multiple monitors.
+	if screenMinPos is None:
+		screenMinPos = wx.Point()
+	# Do not play any tone if offscreen.
+	if 0 <= x <= screenWidth or 0 <= y <= screenHeight:

```

Comment:
It is a matter of style, but personally I'd prefer this to return early.

Reply:
Hi,

Ah, the chained comparison: my message in that fragment was that it should play coordinate beep as long as x and y are referring to things on screen. It was done like this because in browse mode, things will show up offscreen.

Thanks.

Reply:
Yeah, I would also prefer that the logic in this `if` statement was reversed and just had an early return. This reduces the indentation level of everything below.